hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/table/action/rollback/HoodieClientRollbackTestBase.java

```diff
@@ -96,4 +99,61 @@ protected void twoUpsertCommitDataWithTwoPartitions(List<FileSlice> firstPartiti
       assertEquals(1, secondPartitionCommit2FileSlices.size());
     }
   }
+
+  protected void insertOverwriteCommitDataWithTwoPartitions(List<FileSlice> firstPartitionCommit2FileSlices,
+                                                            List<FileSlice> secondPartitionCommit2FileSlices,
+                                                            HoodieWriteConfig cfg,
+                                                            boolean commitSecondInsertOverwrite) throws IOException {
+    //just generate two partitions
+    dataGen = new HoodieTestDataGenerator(new String[]{DEFAULT_FIRST_PARTITION_PATH, DEFAULT_SECOND_PARTITION_PATH});
+    HoodieTestDataGenerator.writePartitionMetadata(fs, new String[]{DEFAULT_FIRST_PARTITION_PATH, DEFAULT_SECOND_PARTITION_PATH}, basePath);
+    SparkRDDWriteClient client = getHoodieWriteClient(cfg);
+    /**
+     * Write 1 (upsert)
+     */
+    String newCommitTime = "001";
+    List<HoodieRecord> records = dataGen.generateInsertsContainsAllPartitions(newCommitTime, 2);
```

Comment:
this code is the same as twoUpsertCommitDataWithTwoPartitions, can we reuse it ?

Reply:
We can only reuse two lines because some of the state generated (List<HoodieRecord>) is needed for subsequent operations. So i'm not inclined to add another method for this. let me know if you have strong opinion.

Reply:
ok,Make sense